pandas/core/internals/blocks.py

```diff
@@ -1043,7 +1053,7 @@ def setitem(self, indexer, value, using_cow: bool = False) -> Block:
             casted = np_can_hold_element(values.dtype, value)
         except LossySetitemError:
             # current dtype cannot store value, coerce to common dtype
-            nb = self.coerce_to_target_dtype(value)
+            nb = self.coerce_to_target_dtype(value, warn_on_upcast=True)
```

Comment:
looks like this is just in Block. dont we need the same thing in EABackedBlock?

Reply:
yup! OK to do that in a separate PR to keep the files changed down?

Reply:
sure